SIL.Core/Extensions/StringExtensions.cs

```diff
@@ -379,5 +379,150 @@ public static string RemoveDiacritics(this string stIn)
 
 			return sb.ToString().Normalize(NormalizationForm.FormC);
 		}
+
+		/// <summary>
+		/// Gets the longest substring that two strings have in common. The substring returned
+		/// will either be one or more contiguous whole words or a substring that is part of a
+		/// single word (if so requested by the caller). In the latter case, the returned
+		/// substring must be at least <paramref name="minPctForPartialWordMatch"/> of the
+		/// total string length to be considered useful.
+		/// </summary>
+		/// <param name="s1">The first string.</param>
+		/// <param name="s2">The other string.</param>
+		/// <param name="foundWholeWords">Indicates whether the substring being returned is one
+		/// or more whole words (undefined if no useful substring is found)</param>
+		/// <param name="minPctForPartialWordMatch">If set to <c>1.0</c> only whole-word
+		/// substrings will be considered useful. Otherwise, this indicates the minimum
+		/// percentage of the shorter string that must be in common between the two in order
+		/// for it to be considered useful.</param>
+		/// <returns>The longest whole-word (or otherwise "useful") substring that two strings
+		/// have in common.</returns>
+		public static string GetLongestUsefulCommonSubstring(this string s1, string s2,
+			out bool foundWholeWords, double minPctForPartialWordMatch = 1.0)
```

Comment:
This method is pretty long and has two parts: finding whole word substring and finding partial word substring, so it might be better to split it into two (private) methods.